src/pfe/portal/modules/User.js

```diff
@@ -213,49 +213,28 @@ module.exports = class User {
    * Function to get watchList for all open projects
    */
   async getWatchList() {
-    let fileNameList = await fs.readdir(this.directories.projects);
     let watchList = {
       projects: []
     };
-    await Promise.all(fileNameList.map(async (fileName) => {
-      let file = path.join(this.directories.projects, fileName);
-      if (!fileName.startsWith('.')) {
-        try {
-          const projFile = await fs.readJson(file);
-          // do not add closed project in the list
-          if (projFile.state == Project.STATES.closed) {
-            return;
-          }
-          const project = {};
-          const projName = projFile.name;
-          project.pathToMonitor = projFile.locOnDisk;
-          const isWindowsPath = cwUtils.isWindowsAbsolutePath(project.pathToMonitor);
-          if (isWindowsPath) {
-            project.pathToMonitor = cwUtils.convertFromWindowsDriveLetter(project.pathToMonitor);
-          }
-          project.projectID = projFile.projectID;
-          project.ignoredPaths = projFile.ignoredPaths;
-          if (projFile.projectWatchStateId == undefined) {
-            project.projectWatchStateId = crypto.randomBytes(16).toString("hex");
-            let projectUpdate = { projectID: projFile.projectID, projectWatchStateId: project.projectWatchStateId };
-            await this.projectList.updateProject(projectUpdate);
-          } else {
-            project.projectWatchStateId = projFile.projectWatchStateId;
-          }
-          project.projectCreationTime = projFile.creationTime;
-          log.debug("Find project " + projName + " with info: " + JSON.stringify(project));
-          watchList.projects.push(project);
-        } catch (err) {
-          // Corrupt project inf file
-          if (err instanceof SyntaxError) {
-            log.error('Failed to parse project file ' + fileName);
-          } else {
-            log.error(err);
-          }
+    let projectArray = this.projectList.getAsArray();
+    for (let i = 0; i < projectArray.length; i++) {
+      // Only include enabled projects
+      if (projectArray[i].isOpen()) {
+        if (projectArray[i].projectWatchStateId == undefined) {
+          let watchStateId = crypto.randomBytes(16).toString("hex");
+          let projectUpdate = { projectID: projectArray[i].projectID, projectWatchStateId: watchStateId };
+          await this.projectList.updateProject(projectUpdate);
+        } 
+        let project = {
+          projectID: projectArray[i].projectID,
```

Comment:
Could use destructuring `const { projectID, projectWatchStateId, pathToMonitor, ignoredPaths, creationTime: projectCreationTime } = projectArray[i]` to simplify